pkg/controller/upload-controller.go

```diff
@@ -426,7 +268,8 @@ func (c *UploadController) getOrCreateUploadPod(pvc *v1.PersistentVolumeClaim, p
 			ClientCA:       clientCA,
 		}
 
-		pod, err = CreateUploadPod(args)
```

Comment:
Since `CreateUploadPod` is a member function now, I think it makes sense to revisit the args structure.  At a minimum, some fields can be removed.  Maybe even get rid of it entirely.

Reply:
Yeah all the fields in the reconciler do not have to get passed anymore.